cpp/ql/test/query-tests/Critical/MissingCheckScanf/test.cpp

```diff
@@ -406,3 +406,26 @@ char *my_string_copy() {
 	*ptr++ = 0;
 	return DST_STRING;
 }
+
+void scan_and_write() {
+	{
+		int i;
+		if (scanf("%d", &i) < 1) {
+			i = 0;
+		}
+		use(i);  // GOOD [FALSE POSITIVE]: variable is overwritten with a default value when scanf fails
+	}
+	{
+		int i;
+		if (scanf("%d", &i) != 1) {
+			i = 0;
+		}
+		use(i);  // GOOD [FALSE POSITIVE]: variable is overwritten with a default value when scanf fails
+	}
+}
+
+void scan_and_static_variable() {
+	static int i;
+	scanf("%d", &i);
+	use(i);  // GOOD [FALSE POSITIVE]: static variables are always 0-initialized
```

Comment:
This one's perhaps a bit of a maybe - while `i` should have been initialized it seems like there's a good chance in real world code we're still using an _incorrect_ value in a case like this.  Still, I gather this is motivated by a real world case where using the zero initialization _is_ OK, so that speaks for itself.  We don't want false positives.

Reply:
The thing is that a static variable will never have a random unexpected value: it will start out with 0, and then it will have whatever value gets written to it during the program execution, as written in the program itself. So yes, it's true this could still be a bug, but not the kind of bug that suddenly makes you try to load 4294967295 entries :laughing: